tests/pysharedutils/dicts_tests.py

```diff
@@ -281,3 +281,100 @@ def test_get_dict_properties_without_strict_and_invalid_property(self):
             'comment.user.username'
         )
         assert_equal(output, {'comment.user.username': None})
+
+
+class TestMapDictKeys:
+
+    def test_map_dict_keys(self):
+        obj = {
+            'first_name': 'Joe',
+            'middle_name': 'P',
+            'last_name': 'Smith'
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'last_name': 'surname',
+            'mis_spelled_key': 'mis_spelled_value'
+        }
+        output = pysharedutils.map_dict_keys(obj, map_obj)
+        expected_output = {
+            'given_name': 'Joe',
+            'middle_name': 'P',
+            'surname': 'Smith'
+        }
+        assert_equal(output, expected_output)
+
+    def test_map_dict_keys_with_one_level(self):
+        obj = {
+            'first_name': 'Joe',
+            'user': {
+                'username': 'joe@example.com'
+            }
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'user.username': 'email',
```

Comment:
I'm confused how the right hand value (`email`) changes the key of the _nested_ dictionary. I thought it would return `{'email': 'joe@example.com'}`, not `{'user': {'email': 'joe@example.com'}}`